src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -49,11 +51,11 @@
 
 public class HaxeSemanticAnnotator implements Annotator {
   @Override
-  public void annotate(PsiElement element, AnnotationHolder holder) {
+  public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder holder) {
     analyzeSingle(element, holder);
   }
 
-  static void analyzeSingle(final PsiElement element, AnnotationHolder holder) {
+  private static void analyzeSingle(final PsiElement element, AnnotationHolder holder) {
```

Comment:
Well, there are several reasons. The main one - classes should not provide more API than needed. Class in case of encapsulation is a black-box, and if some "hands" should stick out - they will, otherwise there is no reason to raise count of unused (except internal needs) API, even package-private is not alawas necessary. Often this methods are confusing the developer, and he may take a wrong choise, or he has got to get into the code and try to realise - why he should use this method or other one. That's my point.